### ISSUE TITLE
Record#save, Mapper#update*, and DataStore#update* do not "commit" changes to records

### ISSUE THREAD

**Issue by jmdobry** [js-data/js-data#340]

**Comment by stalniy** [comment]
Could you please add description of what is wrong? So, then I could fix that.

I believe the issue is that when js-data does crud it doesn't call `commit` method on the record, right?

**Comment by jmdobry** [comment]
> doesn't call `commit` method on the record

That about sums it up. However, I am unsure of the best way to handle this, or whether to handle it at all. My thoughts so far are to make changes to:

* `Record#save`, though the code to parse the returned record will be sophisticated.
* `Collection#add`, via a new option called `commit`, which if `true` will call `commit` on records that are being added to the collection, but already exist in the collection. This will cover the case of DataStore#update*

I think we should let this issue simmer for a bit.

**Comment by k2s** [comment]
It would be big help for me if this would be implemented. For my code it is breaking change between v2 and v3.

I am binding in Aurelia to js-data records. Some actions do update on the data with id and I was used that the bound record will be also changed. 

How I understood the design of v2, js-data provided to me access to "singleton" records. If I added property to record, it was merged to existing record and I had access to the changes in all places where I was referencing to that record object.
